framework/src/play/mvc/Mailer.java

```diff
@@ -540,76 +534,84 @@ public static Future<Boolean> send(Object... args) {
                     }
                 }
             }
-            email.setCharset("utf-8");
+            setAddresses(email);
+            if (!StringUtils.isEmpty(charset)) {
+                email.setCharset(charset);
+            }
 
-            if (from != null) {
-                try {
-                    InternetAddress iAddress = new InternetAddress(from);
-                    email.setFrom(iAddress.getAddress(), iAddress.getPersonal());
-                } catch (Exception e) {
-                    email.setFrom(from);
-                }
+            email.setSubject(subject);
+            email.updateContentType(contentType);
 
+            if (headers != null) {
+                for (String key : headers.keySet()) {
+                    email.addHeader(key, headers.get(key));
+                }
             }
 
-            if (replyTo != null) {
-                try {
-                    InternetAddress iAddress = new InternetAddress(replyTo);
-                    email.addReplyTo(iAddress.getAddress(), iAddress.getPersonal());
-                } catch (Exception e) {
-                    email.addReplyTo(replyTo);
-                }
+            return Mail.send(email);
+        } catch (EmailException ex) {
+            throw new MailException("Cannot send email", ex);
+        }
+    }
 
+    static void setAddresses(Email email) throws EmailException {
+        email.setCharset("utf-8");
```

Comment:
why forced the charset to utf-8 here ? in the upper code you only set the charset if define ?
Do you mean that a default value is need ?

Reply:
@xael-fry  
This encoding fix was originally written, in line 543.  
This line appears to have been moved to the setAddresses () method as it relates to setting addresses.